### DISCUSSION TITLE
Reverse Proxy sub page not loading correctly

### DISCUSSION THREAD

**Discussion by Naomarn** [hrfee/jfa-go#228] [Q&A]
I finally have this working in nginx proxy manager with custom location for a sub folder, however, the page does not load correctly. I have tried many things to try and fix and even used the wiki for reverse proxy settings but I always get the same result. Has anyone else ran into this or have a suggestion on what I can try?
<img width="366" alt="jfa-go error page" src="https://user-images.githubusercontent.com/116525162/197583627-eb6eeb58-34f9-429b-9bd3-154b4689578f.png">

**Comment by ohbe1jacobi** [comment]
Did you ever figure this out, @Naomarn ?

**Reply by Kratos776** [reply]
i do have the same problem. did anyone got ir to work? @Naomarn @jebriggsy ??

**Reply by ohbe1jacobi** [reply]
Sorry @Kratos776 I never got it to work.  I put it on a sub-subdomain instead

**Reply by timbroder** [reply]
Figured I'd check in as I'm running into this with Caddy. any luck?

**Reply by timbroder** [reply]
Nevermind, this worked

```
[my domain] {
        reverse_proxy jellyfin:8096
        reverse_proxy /accounts/* jfago:8056 {
                header_up Host {host}
                header_up X-Real-IP {remote}
                header_up X-Forwarded-For {remote}
                header_up X-Forwarded-Proto {scheme}
                header_down Content-Security-Policy ""
        }
}
```

**Comment by hrfee** [comment]
Had an attempt myself and I cannot figure it out either, sorry. I also made sure the wiki config still works, and it does for me on a sub-folder with normal nginx.

**Reply by jcschauer** [reply]
I'm having the same problem as @Naomarn @Kratos776 @ohbe1jacobi and have followed the wiki and tried and tried... 
I'm hoping someone has a new idea for me.

I am using nginx version: nginx/1.31.2
jfa-go Version 0.6.0 Commit 96983d7  Build Time 2025-11-27 20:46:27 +0000 UTC
Host Linux Ubuntu 24.04.4 LTS
docker-compose version 1.29.2, build unknown

The first curl redirect is expected. All of my subfolders are configured this way. Second one is where it's interesting:

$curl -I https://smoke.redgreen.net/accounts
HTTP/1.1 301 Moved Permanently
Server: nginx/1.31.2
Date: Mon, 06 Jul 2026 18:15:16 GMT
Content-Type: text/html
Content-Length: 169
Location: https://smoke.redgreen.net/accounts/
Connection: keep-alive

$curl -I https://smoke.redgreen.net/accounts/
HTTP/1.1 404 Not Found
Server: nginx/1.31.2
Date: Mon, 06 Jul 2026 18:15:19 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 2009
Connection: keep-alive
Cache-Control: no-cache
Link: </accounts/css/v0.6.0bundle.css>; rel=preload; as=style, </accounts/css/remixicon.css>; rel=preload; as=style

I am running both via docker-compose jellyfin and jfs-go in one, and nginx in another using host networking.


My other sub-page apps work fine with "location /appname/" which is why I tried it both ways and even both at once.
My nginx/conf.d/default.conf  contains:
server {
    listen 443 ssl;
    listen [::]:443 ssl;
    server_name  smoke.redgreen.net;
#...
# jfa-go version 0.6.0
    location ^~ /accounts  {
        # Remove the CSP header set for Jellyfin
        proxy_hide_header Content-Security-Policy;
        add_header Content-Security-Policy "";
        proxy_pass http://192.168.1.251:8056/accounts; # change as you need
        http2_push_preload on; # Should make the page load quicker.
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_buffering off;
    }
    location ^~ /accounts/  {
        # Remove the CSP header set for Jellyfin
        proxy_hide_header Content-Security-Policy;
        add_header Content-Security-Policy "";
        proxy_pass http://192.168.1.251:8056/accounts/; # change as you need
        http2_push_preload on; # Should make the page load quicker.
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_buffering off;
    }



My jellyfin and jfs-go docker-compose.yaml
    version: "3.5"
    services:
      jellyfin:
        image: jellyfin/jellyfin
        container_name: jellyfin
        network_mode: 'host'
        volumes:
          - ./config:/config
          - ./cache:/cache
          - ../certbot/data/certbot/conf:/etc/letsencrypt
          - /x/opt-t:/x/opt-t:ro # Replace with your media directory
        restart: unless-stopped

      jfa-go:
        image: hrfee/jfa-go:latest
        container_name: jfa-go
        network_mode: 'host'
        volumes:
          - ./jfa-config:/data
          - ./config:/jf:ro
        restart: unless-stopped
        depends_on:
          - jellyfin


ngnix docker-compose.yml
version: '3.7'
services:
  nginx:
    image: nginx:alpine # Use a lightweight Nginx image
    ports:
      - "80:80"       # Map host port 80 to container port 80
      - "443:443"
    volumes:
      - ./nginx/conf.d:/etc/nginx/conf.d
      - ./certbot/www:/var/www/certbot
      - ./certbot/conf:/etc/letsencrypt
      - ./html:/usr/share/nginx/html
      - ./logs/:/var/log/nginx:rw
    restart: always

  certbot:
    image: certbot/certbot
    container_name: certbot
    volumes:
      - ./certbot/www:/var/www/certbot
      - ./certbot/conf:/etc/letsencrypt
    depends_on:
      - nginx

###  Update
After all that writeup, I found it seems to be working via firefox, but curl still reports the not found error so I'm leaving my comment there. Hopefully all my configs will help someone else.
